### PR TITLE
[DeviceSanitizer] Support check invalid kernel argument

### PR DESCRIPTION
**Pull request by AllanZyne**
UR: https://github.com/oneapi-src/unified-runtime/pull/1843

also fix parallel_for_double.cpp

### CODE REVIEW

**Review by yingcong-wu — COMMENTED** [review]
out-of-bounds and released-pointer checks seem a bit redundant. I mean those could also be caught when those errors happened within kernel execution, right?

**Comment by AllanZyne** [comment]
> out-of-bounds and released-pointer checks seem a bit redundant. I mean those could also be caught when those errors happened within kernel execution, right?

Yeah, that's a problem. I'll reconsider this.

**Comment by AllanZyne** [comment]
> out-of-bounds and released-pointer checks seem a bit redundant. I mean those could also be caught when those errors happened within kernel execution, right?

Yes, this a bit redundant. But it costs very little to check these two errors while checking other errors.

**Comment by AllanZyne** [comment]
Hi @intel/unified-runtime-reviewers, please review, thanks

File: sycl/test-e2e/AddressSanitizer/out-of-bounds/USM/parallel_for_double.cpp [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 int main() {
   sycl::queue Q;
-  constexpr std::size_t N = 123456;
+  constexpr std::size_t N = 512;
```
Comment (AllanZyne) [inline sycl/test-e2e/AddressSanitizer/out-of-bounds/USM/parallel_for_double.cpp:19]: We'll add a new test in “private” for large workgroup case

**Comment by omarahmed1111** [comment]
@intel/dpcpp-sanitizers-review could you have another round of review on this as I merged the UR part of it, so we would prioritize merging that. Thanks!

**Review by yingcong-wu — APPROVED** [review]

**Review by zhaomaosu — APPROVED** [review]

**Comment by AllanZyne** [comment]
> @intel/dpcpp-sanitizers-review could you have another round of review on this as I merged the UR part of it, so we would prioritize merging that. Thanks!

Sure.

**Comment by AllanZyne** [comment]
@intel/llvm-gatekeepers please merge, thanks.

**Pull request merged**